massifs/bloberrors.go

```diff
@@ -0,0 +1,129 @@
+package massifs
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	azStorageBlob "github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+)
+
+const (
+	azblobBlobNotFound = "BlobNotFound"
+)
+
+func AsStorageError(err error) (azStorageBlob.StorageError, bool) {
```

Comment:
why not use errors.As()? 

do we not wrap `azStroageBlob.StorageError`?

also why not return the pointer? does `*azStorageBlob.StorageError` not implement the error interface?

same question for the below `AsResponseError`

Reply:
In various places in the stack we can have either Internal or Storage and its a total pain to figure that out all the time. On its own the errors.As is a pain to work with in the contexts we need. It takes multiple fiddly steps to do the right thing. This code was previously reviewed and is only "green" because git thinks the file is an adition. I just followed the previously established pattern for AsResponseError and IsRateLimiting. Which are the only functions that have been added or changed.

Reply:
re open for further discussion if you want. But please offer an example of what you mean both here and at the call sites that need to specifically deal with it being a BlobNotFound or a 429

Reply:
We just need to make these receivers on the error types we define, and rename them to align with the interface:

```
func (se *StorageError) As(err error) bool
```

Reply:
Yes please :) 

If this really needs to go in ASAP, it makes the code no worse than it already is and we can raise a TD to do this straight after.

I think this must have fallen through on a previous PR, that is probably me as a reviewer missing it, so thats my bad.

We should really use the golang errors interfaces where we can, as its the standard way of doing this and i don't really understand the reason we shouldn't?

Happy to do some pairing or a catchup call to go over it in more detail :)

Reply:
resolved with as it works and is tested and @robinbryce has raised a td ticket: [User Story 9917](https://dev.azure.com/jitsuin/avid/_workitems/edit/9917): TD: IsBlobError and IsRateLimiting helpers